Framework/PythonInterface/plugins/algorithms/LoadVesuvio.py

```diff
@@ -140,19 +141,72 @@ def PyExec(self):
 #----------------------------------------------------------------------------------------
 
     def validateInputs(self):
+        self._load_common_inst_parameters()
         issues = {}
 
-        # Validtae run number ranges
+        # Validate run number ranges
         run_str = self.getProperty(RUN_PROP).value
         if "-" in run_str:
             lower, upper = run_str.split("-")
-            if upper < lower:
-                issues[RUN_PROP] = "Range must be in format lower-upper"
+            issues = self._validate_range_formatting(upper, lower, RUN_PROP, issues)
+
+        # Validate SpectrumList
+        grp_spectra_list = self.getProperty(SPECTRA_PROP).value
+        if ";" in grp_spectra_list:
+            # Split on ';' if in form of 2-3;6-7
+            grp_spectra_list = grp_spectra_list.split(";")
+        else:
+            # Treat input as a list
+            grp_spectra_list = [grp_spectra_list]
+
+        for spectra_grp in grp_spectra_list:
+            spectra_list = None
+            if "-" in spectra_grp:
+                # Split ranges
+                spectra_list = spectra_grp.split("-")
+                # Validate format
+                issues = self._validate_range_formatting(spectra_list[0], spectra_list[0], SPECTRA_PROP, issues)
+            elif "," in spectra_grp:
+                # Split comma separated lists
+                spectra_list = spectra_grp.split(",")
+            else:
+                # Single spectra
+                spectra_list = spectra_grp
+
+            # Validate boundaries
+            for spec in spectra_list:
+                spec = int(spec)
+                issues = self._validate_spec_min_max(spec, issues)
+
+        return issues
+
+#----------------------------------------------------------------------------------------
 
+    def _validate_range_formatting(self, upper, lower, property_name, issues):
+        """
+        Validates is a range style input is in the correct form of lower-upper
+        """
+        if upper < lower:
+            issues[property_name] = "Range must be in format lower-upper"
         return issues
 
 #----------------------------------------------------------------------------------------
 
+    def _validate_spec_min_max(self, spectra, issues):
+        """
+        Validates if the spectra is with the minimum and maximum boundaries
+        """
+        # Only validate boundaries if in difference Mode
+        if "Difference" in self.getProperty(MODE_PROP).value:
+            specMin = self._backward_spectra_list[0]
+            specMax = self._forward_spectra_list[len(self._forward_spectra_list) - 1]
```

Comment:
The refactorings look good. One minor point here would be that Python would allow you to do `self._forward_spectra_list[-1]` to access the back element rather than having to query the list length. This is generally neater and avoids bugs if you accidentally query the wrong list for size